x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/alert_renderer/index.tsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get } from 'lodash/fp';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import React from 'react';
+import styled from 'styled-components';
+
+import { AlertField } from './alert_field';
+import type { RowRenderer } from '../../../../../../../common/types';
+import { RowRendererId } from '../../../../../../../common/types';
+import {
+  ID,
+  DESTINATION_IP,
+  DESTINATION_PORT,
+  EVENT_CATEGORY,
+  FILE_NAME,
+  HOST_NAME,
+  KIBANA_ALERT_RULE_NAME,
+  KIBANA_ALERT_SEVERITY,
+  PROCESS_NAME,
+  PROCESS_PARENT_NAME,
+  SOURCE_IP,
+  SOURCE_PORT,
+  USER_NAME,
+  WITH_FIELD_NAMES,
+  eventKindMatches,
+  showWith,
+} from './helpers';
+import { Details } from '../helpers';
+import { RowRendererContainer } from '../row_renderer';
+import * as i18n from './translations';
+
+export const DEFAULT_CONTEXT_ID = 'alert-renderer';
+
+const AlertRendererFlexGroup = styled(EuiFlexGroup)`
+  gap: ${({ theme }) => theme.eui.euiSizeXS};
+`;
+
+export const alertRenderer: RowRenderer = {
+  id: RowRendererId.alert,
+  isInstance: (ecs) => eventKindMatches(get('event.kind', ecs)),
+  renderRow: ({ contextId = DEFAULT_CONTEXT_ID, data, isDraggable, timelineId }) => {
+    const eventId = get(ID, data);
+    const destinationIp = get(DESTINATION_IP, data);
+    const destinationPort = get(DESTINATION_PORT, data);
+    const eventCategory = get(EVENT_CATEGORY, data);
+    const fileName = get(FILE_NAME, data);
+    const hostName = get(HOST_NAME, data);
+    const kibanaAlertRuleName = get(KIBANA_ALERT_RULE_NAME, data);
+    const kibanaAlertSeverity = get(KIBANA_ALERT_SEVERITY, data);
+    const processName = get(PROCESS_NAME, data);
+    const processParentName = get(PROCESS_PARENT_NAME, data);
+    const sourceIp = get(SOURCE_IP, data);
+    const sourcePort = get(SOURCE_PORT, data);
+    const userName = get(USER_NAME, data);
+
+    return (
+      <RowRendererContainer>
+        <Details data-test-subj="alertRenderer">
+          <AlertRendererFlexGroup
+            alignItems="center"
+            gutterSize="none"
+            justifyContent="center"
+            wrap={true}
+          >
+            <AlertField
```

Comment:
Would it be easier to just have a config array and just run a `map` for these components? You can just default `contextId, eventId, isDraggable, timelineId` in the single component and toss the rest in a config array?

Reply:
Thanks for paring over zoom re: this comment @michaelolo24!

For the folks playing along at home: we agreed while paring not to change this, because it required the creation of specialized (and therefore unlikely to be reused) code as a result of a few nuances.